File: shadowd/types/device.go
package types

import (
	"time"
)

// Device represents a device in the Mesh network
type Device struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	OS        string    `json:"os"`
	OSVersion string    `json:"os_version"`
	MeshIP    string    `json:"mesh_ip"`
	PublicKey string    `json:"public_key"`
	IsOnline  bool      `json:"is_online"`
	LastSeen  time.Time `json:"last_seen"`
	SSHPort   int       `json:"ssh_port"`
	GRPCPort  int       `json:"grpc_port"`
}

// PairingCode represents the data encoded in a pairing QR code
type PairingCode struct {
	DeviceID   string `json:"device_id"`
	DeviceName string `json:"device_name"`
	MeshIP     string `json:"mesh_ip"`
	PublicKey  string `json:"public_key"`
	Timestamp  int64  `json:"timestamp"`
}

// HealthStatus represents the health status of the daemon
type HealthStatus struct {
	Status    string    `json:"status"`
	Uptime    int64     `json:"uptime"`
	Connected bool      `json:"connected"`
	LastCheck time.Time `json:"last_check"`
}

// ConnectionStatus represents the connection status to Headscale
type ConnectionStatus struct {
	Connected      bool      `json:"connected"`
	HeadscaleURL   string    `json:"headscale_url"`
	LastConnected  time.Time `json:"last_connected"`
	ReconnectCount int       `json:"reconnect_count"`
}

File: mobile-app/src/types/device.ts
/**
 * Device types for Shadow Shuttle mobile app
 */

export interface Device {
  id: string;
  name: string;
  hostname: string;
  os: string;
  meshIP: string;
  sshPort: number;
  grpcPort: number;
  publicKey: string;
  online: boolean;
  lastSeen: Date;
}

export interface PairingCode {
  deviceId: string;
  meshIP: string;
  sshPort: number;
  grpcPort: number;
  publicKey: string;
  timestamp: number;
  signature: string;
}

export interface ConnectionStatus {
  connected: boolean;
  meshIP?: string;
  error?: string;
}

export interface HealthStatus {
  healthy: boolean;
  uptime: number;
  lastHeartbeat: Date;
}
